File: java/2.x/src/com/relevantcodes/extentreports/CategorySourceBuilder.java
/*
* Copyright (c) 2015, Anshoo Arora (Relevant Codes).  All rights reserved.
* 
* Copyrights licensed under the New BSD License.
* 
* See the accompanying LICENSE file for terms.
*/

package com.relevantcodes.extentreports;

import java.util.Collections;
import java.util.List;

import com.relevantcodes.extentreports.source.CategoryHtml;
import com.relevantcodes.extentreports.source.ExtentFlag;

public class CategorySourceBuilder {
    public static String buildOptions(List<String> categories) {
        String source = "";
        
        categories = categories.subList(0, categories.size());
        Collections.sort(categories);
        
        for (String c : categories) {
            source += CategoryHtml.getOptionSource()
                    .replace(ExtentFlag.getPlaceHolder("testCategory"), c)
                    .replace(ExtentFlag.getPlaceHolder("testCategoryU"), c.toLowerCase().replace(" ", ""));
        }
        
        return source;
    }
}

File: dotnet/ExtentReports/CategorySourceBuilder.cs
// ***********************************************************************
// Copyright (c) 2015, Anshoo Arora (Relevant Codes). All rights reserved.
//
// Copyrights licensed under the New BSD License.
//
// See the accompanying LICENSE file for terms.
// ***********************************************************************

namespace RelevantCodes.ExtentReports
{
    using System;
    using System.Collections.Generic;
    using System.Linq;
    using System.Text;

    using Model;
    using Source;

    internal class CategorySourceBuilder {
        public static string buildOptions(List<string> Categories) {
            string source = "";

            Categories.Sort();

            var catFlags = new string[] { 
                ExtentFlag.GetPlaceHolder("testCategory"), 
                ExtentFlag.GetPlaceHolder("testCategoryU") 
            };

            Categories.ForEach(c =>
            {
                var catValues = new string[] { 
                    c, 
                    c.ToLower().Replace(" ", "") 
                };

                source += SourceBuilder.Build(CategoryHtml.GetOptionSource(), catFlags, catValues);
            });
        
            return source;
        }
    }
}
